### PR TITLE
Grep Implementations and Misc. Practice Scripts

### PR DESCRIPTION
**Pull request by KevinShimotakahara**
Done these tickets (Sorry for the backlog of code, I forgot I did a bunch of practice code tickets): 

https://www.notion.so/Lambda-Stream-APIs-cefd7012bdb940e99ff705f5e5347dbf
https://www.notion.so/Learning-Lambda-and-Stream-APIs-573c03e1cef44ce48fa3005b052216eb
https://www.notion.so/Design-Implementation-bc4c445eebc24acf8d63d5f260213d2d
https://www.notion.so/Java-Regex-21ac72ce8d8640799c0b36c907f6e6bc

### CODE REVIEW

**Comment by KevinShimotakahara** [comment]
In the lambda grep implementation I am using an API that is only for Java 11, so I will need to re-write a method using what's available in Java 8. I just noticed that this is a problem when I was trying to compile the project with Maven in the command line. I assume I could just edit the Maven file to use Java 11, but I think it would be good practice for me to know how to implement the methods with Java 8.

File: core_java/grep/log/grep.out [hunk 1]
```diff
@@ -0,0 +1,8 @@
+    Is father, mother, Tybalt, Romeo, Juliet,
+Enter Romeo and Juliet aloft, at the Window.
+    And Romeo dead; and Juliet, dead before,
```
Comment (jarviscanada) [inline core_java/grep/log/grep.out]: git rm log

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+//    try {
+//      while ((line = myReader.readLine()) != null) {
+//        String data = line;
+//        fileLines.add(data);
+//      }
+//      myReader.close();
+//    } catch(IOException ex){
+//      ex.printStackTrace();
+//    }
+//    return fileLines;
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: clean up

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+   * Top level search workflow
+   * @throws IOException
+   */
+  public void process() throws IOException {
+    try {
+      FileWriter fw = new FileWriter(outFile, true);
+      BufferedWriter bw = new BufferedWriter(fw);
+      this.out = new PrintWriter(bw);
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: use `logger.error("meaningful mesage", e)`..never print out your log

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+//      javaGrepLambdaImp.setRootPath(args[1]);
+//      javaGrepLambdaImp.setOutFile(args[2]);
+    JavaGrepImp javaGrepLambdaImp = new JavaGrepLambdaImp(args[0],args[1],args[2]);
+
+    try {
+      //calling parent method,
+      //but it will call override method (in this class)
+      javaGrepLambdaImp.process();
+    } catch (Exception ex){
+      ex.printStackTrace();
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java:43]: remove all printstacktrack..and use logger instead

File: core_java/src/main/java/ca/jrvs/practice/HelloWorld.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package ca.jrvs.apps.practice;
+
+class HelloWorld {
+
+  // Your program begins with a call to main().
```
Comment (jarviscanada) [inline core_java/src/main/java/ca/jrvs/practice/HelloWorld.java:5]: remove

Comment (jarviscanada) [reply core_java/src/main/java/ca/jrvs/practice/HelloWorld.java:5]: remove this file

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+//      e.printStackTrace();
+//    }
+//    return response;
+//  }
+
+//  @Override
+//  public HttpResponse httpGet(URI uri) {
+//    HttpGet request = new HttpGet(uri);
+//    HttpResponse response = null;
+//    try {
```
Comment (jarviscanada) [inline core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java]: clean up

**Review by jarviscanada — APPROVED** [review]

**Pull request merged**